### PR TITLE
ft #161405965 Tests for API Endpoints with Authentication

### PR DESCRIPTION
**Pull request by Nicanor008**
**What does this PR do?**
Have tests for all endpoints . A specific users access a specific endpoint that he/she is authorised to.

**Description of the Task**
- Written user registration and login test endpoint
- By use of tokens to help identify a specific user, I have written tests for:
1.Fetching products
2.Posting posting
3.Selecting a specific product 
4.Fetching sales
5.Posting sales
6.Fetching a single sale 

**How to manually test**
Clone this repo
 `git clone https://github.com/Nicanor008/Store-Manager-API.git`
`cd Store-Manager-API`
Activate the virtualenv `source venv/bin/activate`
install the reuired dependecies `pip install -r requirements.txt`
Checkout to this branch `git checkout ft-test-APIEndpoints-WithAuthentication-161405965`
Run all tests `nosetests app/tests/v1` or just `nosetests`

**Any Background context**
Null

**What are the relevant Pivotal tracker stories**
#161405965

### CODE REVIEW

File: app/tests/v1/test_products.py [hunk 1]
```diff
@@ -1,73 +1,116 @@
-import unittest 
 import json
-from app import create_app
+from app.tests.v1.base_tests import BaseTest
```
Comment (Bonifase) [inline app/tests/v1/test_products.py:2]: Is there a reason for having the folder V1 inside your tests folder? And there are additional base tests files in V1 folder, what is their purpose?

Comment (Nicanor008) [reply app/tests/v1/test_products.py:2]: I had got an error initially and it was asking on those files, (base_testing.py), have consulted on how to deal with it. let me remove. The v1 only tests for challenge 2 in challenge 3 I will use v2 for tests

Comment (Bonifase) [reply app/tests/v1/test_products.py:2]: You will need to have a new repo for challenge 3

Comment (Nicanor008) [reply app/tests/v1/test_products.py:2]: I didn't know that challenge 3 was to be done in another repo, I had implemented as per the guidance I was given. Challenge 3 was supposed to be in v2 as per information I got. So, let me get clear from you, I am supposed to only have app/tests/test_products.py .... and so on files, there should be no V1?

Comment (Nicanor008) [reply app/tests/v1/test_products.py:2]: I have resolved the unnecessary files base_test.py and base_testing.py

Comment (Bonifase) [reply app/tests/v1/test_products.py:2]: cool, you can merge

**Pull request merged**